### ISSUE TITLE
found with fuzzer: hiding under nothing?

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#132]
```
Suddenly, the dungeon collapses.
hiding under nothing?
Generating more information you may report:


Program received signal SIGABRT, Aborted.
__pthread_kill_implementation (no_tid=0, signo=6, threadid=140630863292352) at ./nptl/pthread_kill.c:44
44	./nptl/pthread_kill.c: No such file or directory.
(rr) bt
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=140630863292352) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=140630863292352) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=140630863292352, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007fe72cbc0476 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007fe72cba67f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x0000562b2925902e in NH_abort () at end.c:236
#6  0x0000562b2925ff27 in panic (str=0x562b29d672e0 "%s") at end.c:802
#7  0x0000562b297c0543 in impossible (s=0x562b29d0ada0 "hiding under nothing?") at pline.c:518
#8  0x0000562b294b919e in mattacku (mtmp=0x6110000ad380) at mhitu.c:768
#9  0x0000562b29666947 in dochug (mtmp=0x6110000ad380) at monmove.c:998
#10 0x0000562b29650af9 in dochugw (mtmp=0x6110000ad380) at monmove.c:126
#11 0x0000562b295bd1e8 in movemon () at mon.c:1567
#12 0x0000562b28f6823e in moveloop (resuming=0 '\000') at allmain.c:244
#13 0x0000562b29bf2091 in main (argc=0, argv=0x7ffe1cefe7c8) at ../sys/unix/unixmain.c:353



#6  0x0000562b2925ff27 in panic (str=0x562b29d672e0 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7ffe1cefd690, 
            reg_save_area = 0x7ffe1cefd5c0}}
#7  0x0000562b297c0543 in impossible (s=0x562b29d0ada0 "hiding under nothing?") at pline.c:518
        pbuf = "hiding under nothing?", '\000' <repeats 196 times>...
        the_args = {{gp_offset = 8, fp_offset = 48, overflow_arg_area = 0x7ffe1cefdd80, 
            reg_save_area = 0x7ffe1cefdca0}}
#8  0x0000562b294b919e in mattacku (mtmp=0x6110000ad380) at mhitu.c:768
        obj = 0x0
        mattk = 0x0
        alt_attk = {aatyp = 0 '\000', adtyp = 0 '\000', damn = 0 '\000', damd = 0 '\000'}
        i = 0
        j = 0
        k = 0
        tmp = 0
        sum = {0, 0, 0, 0, 0, 0}
        mdat = 0x562b29e51b40 <mons+52000>
        marmf = 0x0
        ranged = 0 '\000'
        range2 = 0 '\000'
        foundyou = 1 '\001'
        youseeit = 1 '\001'
        skipnonmagc = 0 '\000'
#9  0x0000562b29666947 in dochug (mtmp=0x6110000ad380) at monmove.c:998
        mdat = 0x562b29e51b40 <mons+52000>
        tmp = <optimized out>
        mdummy = 0x0
        inrange = 1
        nearby = 1
        scared = 0
        oldx = 0
        oldy = 0
        mwalk_sewage = 0 '\000'
#10 0x0000562b29650af9 in dochugw (mtmp=0x6110000ad380) at monmove.c:126
        x = 28
        y = 6
        already_saw_mon = 0 '\000'
        rd = 24848
#11 0x0000562b295bd1e8 in movemon () at mon.c:1567
        mtmp = 0x6110000ad380
        nmtmp = 0x6110000c8040
        somebody_can_move = 0 '\000'
#12 0x0000562b28f6823e in moveloop (resuming=0 '\000') at allmain.c:244
        moveamt = 9
        wtcap = 5
        change = 0
        monscanmove = 0 '\000'
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#13 0x0000562b29bf2091 in main (argc=0, argv=0x7ffe1cefe7c8) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
I am able to reproduce this:

1. Create a pit on a regular room
2. Summon a giant with a boulder
3. Go into the pit, drop all items
4. Polymorph into a cobra (or other concealing monster)
5. Make sure we are concealed (our glyph should disappear, and our stat page should show what item we are hiding under.
6. Let the giant throw the boulder into the pit, filling it and burying the items. (We should remain hidden here)
7. Let a monster/giant attack us. This is where the crash occurs.

**Comment by elunna** [comment]
When a boulder fills a hole, it called bury_objs - this already called maybe_unhide_at, so not sure why we would remain hidden after the pit is filled.

https://github.com/elunna/hackem/blob/74e04524d034381349d364c4c38abcddcaba3d5e/src/dig.c#L2061

**Comment by elunna** [comment]
Note: The boulder also seems to have to miss us - otherwise if it hits us I think we get unhidden.

The problem: When the boulder fills the pit, the check for maybe_unhide_at hits this line:
```
if (mtmp && mtmp->mundetected && hides_under(mtmp->data))
```
At this point:
 u.uundetected = 1
but 
mtmp->mundetected = 0

Therefore we are not unhidden.

**Comment by elunna** [comment]
Fixed in commit 081c22aac.
